### PR TITLE
[SYCLomatic] Support memory APIs migration with syclcompat

### PR DESCRIPTION
**Pull request by ziranzha**

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/lib/DPCT/CallExprRewriterMemory.cpp [hunk 1]
```diff
@@ -12,6 +12,32 @@
...
+    OS << "dpct_";
+  }
+  OS << RawName;
+  return Result;
+}
+
+std::string MemoryMigrationRule::getMemoryHelperFunctionName(StringRef Name) {
+  return dpct::getMemoryHelperFunctionName(Name);
+}
+
```
Comment (zhimingwang36) [inline clang/lib/DPCT/CallExprRewriterMemory.cpp:46]: Please add lit test for this PR.

File: clang/test/dpct/usm-syclcompat.cu [hunk 2]
```diff
@@ -0,0 +1,1090 @@
+// FIXME
```
Comment (zhimingwang36) [inline clang/test/dpct/usm-syclcompat.cu]: why FIXME in Line1?

File: clang/test/dpct/usm-syclcompat.cu [hunk 3]
```diff
@@ -0,0 +1,1090 @@
...
+// UNSUPPORTED: system-windows
```
Comment (zhimingwang36) [inline clang/test/dpct/usm-syclcompat.cu:1]: Have you built the migrated result for the 2 lit cases?

Comment (ziranzha) [reply clang/test/dpct/usm-syclcompat.cu:1]: Yes.

File: clang/test/dpct/usm-syclcompat.cu (hunk 2 above)
Comment (ziranzha) [reply clang/test/dpct/usm-syclcompat.cu]: Copy from the original lit test. Will remove it.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**